### DISCUSSION TITLE
Does Directus automatically attach the OS's time zone to dates before storing them into DB?

### DISCUSSION THREAD

**Discussion by powerseed** [directus/directus#14092] [Archive]
Directus version: 9.4.2
In my use case, the dates entered by users are in a certain time zone (for example `Europe/Vienna`), but all dates in the database are in time zone UTC, so I want to create a Directus API hook to convert dates from `Europe/Vienna` to UTC before storing them in DB.

In the Directus API hook, before returning the converted date, I print it out. 

For example, user's input is 2022-07-02T**03**:00:00 in `Europe/Vienna`, 
after conversion, the printed output is `input.my_date = 2022-07-02T05:00:00`, which is in UTC and correct. 

So, I expect this exact 2022-07-02T**05**:00:00 string to be stored in DB.

However, the value ended up in the DB is 2022-07-02T**01**:00:00, where there is an offset of 4 hours, which is just the offset between my OS' time zone (America/Toronto) and UTC. 

My DB's time zone is already set to be in UTC as shown below:
![image](https://user-images.githubusercontent.com/42278237/175582046-f7a024b8-d43a-47b0-8137-7d314fde9fa1.png)

So I guess something happened in Directus: **it may attach the OS' time zone to a date before storing it into DB**, which is not what I want because the date should be in UTC instead of the OS' time zone.

Could someone tell me if this is the case?

If yes, can I turn it off or change Directus's default time zone?
Thanks!

**Comment by licitdev** [comment]
There were inconsistencies of datetime / timestamp prior to `v9.9.0` which was patched through https://github.com/directus/directus/pull/10956. 
Please consider updating to a newer release. https://github.com/directus/directus/releases

If you require timezone information, you will have to use the `timestamp` field type.
Other datetime field types are unaffected by timezones.

**Reply by powerseed** [reply]
Got it. Thanks!
